maas-region/src/charm.py

```diff
@@ -53,6 +55,7 @@
 ]
 
 
+@trace_charm(tracing_endpoint="charm_tracing_endpoint")
```

Comment:
What about `extra_types` on the classes we setup for our integrations? `COSAgentProvider`, `MaasRegionProvider`, `DatabaseRequires`, . Did you try to add them and observe any differences in the captured traces?

Reply:
By adding those classes to the `extra_types` I can see lots of more traces. I've added all of them together with `MaasHelper` that can be useful for inspecting the stacktrace when some error happens (see attached image).
![Screenshot from 2024-10-07 11-04-54](https://github.com/user-attachments/assets/ecddf11a-b2f2-494e-8739-72f79f6cfc41)